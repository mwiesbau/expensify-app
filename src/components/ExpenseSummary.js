import React from 'react'
import { connect } from 'react-redux'
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'
import numeral from 'numeral'

export const ExpenseSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'

  return (
    <h1>Viewing {expenseCount} {expenseWord} totalling {numeral(expensesTotal / 100).format("$0,0.00")}</h1>
  )
}


const mapStateToProps = (state) => {
  const selectedExpenses = selectExpenses(state.expenses, state.filters)
  return {
    expensesTotal: selectExpensesTotal(selectedExpenses),
    expenseCount: selectedExpenses.length
  }
}

export default connect(mapStateToProps)(ExpenseSummary)