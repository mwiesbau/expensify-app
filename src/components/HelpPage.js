import React from 'react'

const HelpPage = () => (
  <p>This is my HelpPage</p>
)

export default HelpPage