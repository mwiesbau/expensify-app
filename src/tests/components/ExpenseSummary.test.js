import React from 'react'
import { shallow } from 'enzyme'
import { ExpenseSummary } from '../../components/ExpenseSummary'
import expenses from '../fixtures/expenses'

test("viewing 1 expense", () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={2.54} />)
  expect(wrapper).toMatchSnapshot()
})


test("viewing 2 expense", () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={2} expensesTotal={16.60} />)
  expect(wrapper).toMatchSnapshot()
})