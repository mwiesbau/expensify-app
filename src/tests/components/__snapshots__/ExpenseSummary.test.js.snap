// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viewing 1 expense 1`] = `
<h1>
  Viewing 
  1
   
  expense
   totalling 
  $0.03
</h1>
`;

exports[`viewing 2 expense 1`] = `
<h1>
  Viewing 
  2
   
  expenses
   totalling 
  $0.17
</h1>
`;
